# Notebook – Faysal
## Project: Voorspelling Jeugdpopulatie Almere (CBS Wijken & Buurten)

**Doel van dit notebook:** 
In dit notebook controleer ik of de opgeschoonde dataset (df_v7_clean.csv) geschikt is voor verdere analyse en modellering.


Binnen fase 1 richt ik mij op:
- Controle van datatypes en structuur
- Controle op missende en onlogische waarden
- Controle op wijkindeling per jaar
- Aanmaken van verhoudingsvariabelen (% jeugd t.o.v. totaal inwoners)
- Basisbeschrijvende statistiek
- Opslaan van een analyse-klare dataset (df_v8_ready_for_analysis.csv)

Het doel is om te waarborgen dat de dataset inhoudelijk en technisch correct is voordat verdere data-analyse en modellering plaatsvinden.

In [4]:
import pandas as pd
import numpy as np

df = pd.read_csv("../../data/processed/df_v5_clean_final.csv")

df.head()

,Peildatum,Wijk,AantalInwoners_5,k_0Tot15Jaar_8,k_15Tot25Jaar_9
0,2018-01-01,WK003401,22790,3975,2420
1,2019-01-01,WK003401,22875,3945,2415
2,2020-01-01,WK003401,23275,4005,2485
3,2021-01-01,WK003401,23530,4180,2465
4,2022-01-01,WK003401,1635,175,150


**Toelichting**
De dataset bevat per rij een combinatie van:
- Peildatum (jaar van meting)
- Wijk (wijkcode)
- Totaal aantal inwoners
- Aantal inwoners 0–15 jaar
- Aantal inwoners 15–25 jaar

Elke rij representeert één wijk in één specifiek jaar.
Dit sluit aan bij de businessvraag, waarin ontwikkeling per wijk en per leeftijdscategorie centraal staat.

De kolommen zijn logisch opgebouwd en bevatten uitsluitend de variabelen die relevant zijn voor het voorspellen van de jeugdpopulatie.

In [6]:
df.info()
df.describe()
df.isnull().sum()

<class 'pandas.DataFrame'>
RangeIndex: 237 entries, 0 to 236
Data columns (total 5 columns):
 #   Column            Non-Null Count  Dtype
---  ------            --------------  -----
 0   Peildatum         237 non-null    str  
 1   Wijk              237 non-null    str  
 2   AantalInwoners_5  237 non-null    int64
 3   k_0Tot15Jaar_8    237 non-null    int64
 4   k_15Tot25Jaar_9   237 non-null    int64
dtypes: int64(3), str(2)
memory usage: 9.4 KB


Peildatum           0
Wijk                0
AantalInwoners_5    0
k_0Tot15Jaar_8      0
k_15Tot25Jaar_9     0
dtype: int64

***Toelichting op structuur- en kwaliteitscontrole***
1. **Structuur en datatypes (df.info())**

Uit de structuurcontrole blijkt dat de dataset 237 observaties bevat met 5 kolommen:
- Peildatum (string)
- Wijk (string)
- AantalInwoners_5 (int64)
- k_0Tot15Jaar_8 (int64)
- k_15Tot25Jaar_9 (int64)

De numerieke variabelen zijn correct als integer opgeslagen. Dit is passend, aangezien het hier om aantallen personen gaat.

De kolom Peildatum is momenteel nog een string en zal in een volgende stap worden omgezet naar een datetime-formaat indien nodig voor tijdsanalyses.

De dataset bevat uitsluitend wijkniveau-data en sluit aan bij de doelstelling om jeugdontwikkeling per wijk te analyseren.

2. **Beschrijvende statistiek (df.describe())**

De beschrijvende statistiek geeft inzicht in de spreiding en orde van grootte van de variabelen:
- Het totaal aantal inwoners per wijk varieert aanzienlijk, wat duidt op verschillen in wijkgrootte.
- Het aantal jongeren (0–15 en 15–25 jaar) volgt logisch uit het totaal aantal inwoners.
- Er zijn geen negatieve waarden zichtbaar in de statistische samenvatting.

Dit bevestigt dat de dataset op hoofdlijnen logisch en consistent oogt.

Voor de businessvraag betekent dit dat zowel absolute aantallen als relatieve verschillen tussen wijken relevant zijn voor verdere analyse.

3. **Controle op missende waarden (df.isnull().sum())**

De controle op missende waarden toont dat er geen ontbrekende waarden aanwezig zijn in de dataset.
Dit betekent dat:
- Er geen imputatie nodig is.
- Er geen rijen verwijderd hoeven te worden.
- De dataset volledig is voor de geselecteerde variabelen.

Vanuit data quality-perspectief is dit positief en maakt dit de dataset geschikt voor verdere analyse en modellering.

In [7]:
df[df["k_0Tot15Jaar_8"] < 0]
df[df["k_15Tot25Jaar_9"] < 0]

df[df["k_0Tot15Jaar_8"] > df["AantalInwoners_5"]]
df[df["k_15Tot25Jaar_9"] > df["AantalInwoners_5"]]

,Peildatum,Wijk,AantalInwoners_5,k_0Tot15Jaar_8,k_15Tot25Jaar_9
